### ISSUE TITLE
RFC: Jupyter/IPython Notebook Features

### ISSUE THREAD

**Issue by biasmv** [biasmv/pv#125]
Hi All,

I've pushed an initial version of an alternative Jupiter/IPython notebook module for PV to master. It's a little more powerful than the previous version and allows to load structures from local files. In theory everything that can be converted to a PDB string can be loaded in the viewer. Later one, one could think of using a richer serialisation format.

The module only supports the basics: creating the viewer, adding molecules to it. For example:

```python
from pvviewer.viewer import *
from pvviewer.mol import Mol
structure = Mol.from_pdb_id('2por')
viewer = Viewer()
viewer.cartoon('porin', structure)
viewer.auto_zoom()
viewer
```

This will load the PDB 2por from PDB.org and display it in the viewer using the cartoon render mode.

@aebrahim, @hainm: Since you seem to be using Jupyter/IPython Notebooks, I thought I might ping you regarding the recent changes.

What would the most important features to you?

**Comment by biasmv** [comment]
Here is the new API, the above sample probably doesn't work anymore :).

```python
import pvviewer as pv

structure = pv.mol.from_pdb_id('2por')
viewer = pv.Viewer()
viewer.cartoon('porin', structure, color=pv.color.ss_succession())
viewer.auto_zoom()
viewer.show()
```

**Comment by hainm** [comment]
@biasmv great, thanks for the ping. 

> What would the most important features to you?

how's about rendering a trajectory from simulation? like [chemview package] (https://github.com/gabrielelanaro/chemview) here

![](https://cloud.githubusercontent.com/assets/4451957/10738243/ca18ba54-7bec-11e5-8564-d0692580f85d.png)

We have an engine, [pytraj/cpptraj] (https://github.com/Amber-MD/pytraj), can parse different types of format from different simulation packages, it would be really great if there is an way to plug our to your.

**Comment by biasmv** [comment]
That sounds like a plan. I'll look into it to see what is required to making that happen.

**Comment by satary** [comment]
Hello @biasmv!
Thank you for your work, looks amazing!
Are there any prerequisites for installation? 
I've used setup.py to install pv, there were no errors, it imports and executes fine (no errors in Jupyter server). The problem is - it creates an empty canvas and that's it. Am I doing something conceptually wrong?
![image](https://cloud.githubusercontent.com/assets/2101895/16109769/2b37cad6-33b3-11e6-94cd-0c4acb315dcb.png)

**Comment by hainm** [comment]
may be you can right click then choose "Inspect", then choose Console to see error message from Javascript?

**Comment by ajfarkas** [comment]
I suspect that you're getting an error for calling `viewer.show()` without
an argument. If you're trying to show the whole model, you shouldn't need
to call `show`.

**Comment by biasmv** [comment]
@satary: It's hard to say what's going on. As @hainm pointed out, you might see what's wrong by taking a look at the JavaScript console of your browser. 

@ajfarkas: The `viewer.show()` above is part of the Python API. It's doesn't work the same way as the JS show you are thinking of. The show call above essentially spits out a bunch of html and JS code that then gets inserted into the DOM :).

**Comment by satary** [comment]
Thanks @hainm!
It sure is a Javascript issue.
> Refused to execute script from 'http://raw.githubusercontent.com/biasmv/pv/master/bio-pv.min.js?v=20160616195731' because its MIME type ('text/plain') is not executable, and strict MIME type checking is enabled.
Uncaught Error: Script error for "pv"
http://requirejs.org/docs/errors.html#scripterror

I don't know if it is google chrome issue or something else

**Comment by biasmv** [comment]
@satary: I've update the python module to use rawgit. Can you try if the latest version is working for you? 

Best,
Marco

**Comment by hainm** [comment]
@biasmv may be let `pv` to use local js file?

**Comment by biasmv** [comment]
@hainm: yes, that's certainly the better solution. I'll look into it. Won't happen before the weekend, though.
